hpat/tests/tests_perf/test_perf_series_str.py

```diff
@@ -0,0 +1,383 @@
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+import itertools
+import os
+import time
+import unittest
+from contextlib import contextmanager
+
+import pandas as pd
+
+import hpat
+from hpat.tests.tests_perf.test_perf_utils import *
+
+
+STRIP_CASES = [
+    'ascii',
+    'tú quiénc te crees?',
+    '大处 着眼，c小处着手c。大大c大处'
+]
+
+
+def usecase_series_len(input_data):
+    start_time = time.time()
+
+    input_data.str.len()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_capitalize(input_data):
+    start_time = time.time()
+
+    input_data.str.capitalize()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_lower(input_data):
+    start_time = time.time()
+
+    input_data.str.lower()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_swapcase(input_data):
+    start_time = time.time()
+
+    input_data.str.swapcase()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_title(input_data):
+    start_time = time.time()
+
+    input_data.str.title()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_upper(input_data):
+    start_time = time.time()
+
+    input_data.str.upper()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_lstrip(input_data):
+    start_time = time.time()
+
+    input_data.str.lstrip()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_rstrip(input_data):
+    start_time = time.time()
+
+    input_data.str.rstrip()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_strip(input_data):
+    start_time = time.time()
+
+    input_data.str.strip()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+@contextmanager
+def do_jit(f):
```

Comment:
Don't understand the idea. Why we need this?

Reply:
I created the context manager to measure compile time several times. Then calculate median of the results. The context manager decorates the function via `hpat.jit` then deletes it from memory to drop cache. Example of usage:
```
for _ in range(5):
    with do_jit(pyfunc) as cfunc:
        compile_time = calc_compile_time(cfunc, data)
        compile_times.append(compile_time)
```

The key idea is to automatic remove compilation results (cache) from memory.